CodingStandard.md

```diff
@@ -0,0 +1,1425 @@
+#includes of other headers can make maintenance difficult and introduce
+subtle issues.
+
+
+**Memory**
+
+A function must not return a pointer or reference to memory allocated
+within that function with temporary or automatic storage duration.
+
+Classes that allocate memory with dynamic storage duration, including
+any memory allocated with the new operator, or allocate other shared resources
+must explicitly define copy constructors and copy assignment operators.
+
+
+**Internal/External Linkage**
+
+Variables with static storage duration that are not intended to be
+referenced outside of their compilation unit must be declared with internal
+linkage.
+
+Variables (and functions and types) declared and used within a single
+compilation unit should be declared inside an anonymous namespace, instead of
+using the `static` keyword.
+
+**Casting**
+
+C++ style casts must be used. Other cast formats (such as C-style
+casts) must not be used. Note that overloading typecast operators in classes is
+allowed.
+
+**Const**
+
+Variables and class member functions must be qualified with the const
+keyword where possible (with consideration to future changes). Exception:
+Function parameter variables passed by value.
+
+**Assignment Operator**
+
+User defined copy assignment operators must always return a reference to
+`*this`.
+
+User defined copy assignment operators must be self-assignment safe.
+
+User defined copy assignment operators in derived classes must explicitly
+perform the assignment of its base class.
+
+**Constructors**
+
+Constructor initializer lists must be used instead of assignment
+initialization in the constructor's body when possible.
+
+**Virtual Destructors**
+
+A class destructor must be virtual if and only if the class is
+polymorphic (that is, it contains virtual methods).
+
+**Operator Overloading**
+
+Logical AND ("&&"), logical OR ("||") and the comma (",") operators
+must never be overloaded.
+
+**Derived Classes**
+
+A derived class must never redefine a non-virtual function.
+
+**Downcasting**
+
+Downcasting must always be given justification.  Generally, a downcast represents 
+a problematic design.
+
+**Return Types**
+
+Returning a non-const reference or pointer to non-public member data from a
+public class member function requires justification.
+
+**Function/Operator Overloading**
+
+Function or operator overloading must only be used when performing the same
+logical function.
+
+**Inheritance**
+
+Behavioral type consistency must be maintained between publicly derived
+classes and base classes. That is, public inheritance must implement an
+"is-a" relationship and subtypes (publicly derived classes) must be
+substitutable.
+
+Use of multiple inheritance or virtual inheritance (that is, using the virtual
+keyword in the inheritance specification for a class to make one or more of its
+parent classes be virtual base classes) requires justification.
+
+**Complexity**
+
+Code functions must not be overly complex.
```

Comment:
There is a potential for more specific wording here.

Reply:
Agreed, but I wasn't sure what to say; this is literally all the words that were in this section of the standard I copied.

Reply:
I like that suggestion.